hazelcast-client/src/test/java/com/hazelcast/client/map/ClientMapWithIndexCreationTest.java

```diff
@@ -0,0 +1,100 @@
+ * limitations under the License.
+ */
+
+package com.hazelcast.client.map;
+
+import com.hazelcast.client.config.ClientConfig;
+import com.hazelcast.client.test.TestHazelcastFactory;
+import com.hazelcast.client.util.StaticLB;
+import com.hazelcast.config.Config;
+import com.hazelcast.config.MapConfig;
+import com.hazelcast.config.MapIndexConfig;
+import com.hazelcast.config.XmlConfigBuilder;
+import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.core.IMap;
+import com.hazelcast.core.Member;
+import com.hazelcast.query.SampleObjects;
+import com.hazelcast.test.HazelcastParallelClassRunner;
+import com.hazelcast.test.HazelcastTestSupport;
+import com.hazelcast.test.annotation.ParallelTest;
+import com.hazelcast.test.annotation.QuickTest;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+
+import java.util.List;
+
+@RunWith(HazelcastParallelClassRunner.class)
+@Category({QuickTest.class, ParallelTest.class})
+public class ClientMapWithIndexCreationTest extends HazelcastTestSupport {
+
+    /**
+     * Given a two members (A, B) cluster, a non-smart client connected to B attempts to create a map proxy targeting member A.
+     */
+    @Test
+    public void test_createMapWithIndexes_whenProxyCreatedOnMemberOtherThanClientOwner() {
+        TestHazelcastFactory factory = new TestHazelcastFactory();
+
+        Config config = new XmlConfigBuilder().build();
+
+        MapConfig mapConfig = config.getMapConfig("test");
+        List<MapIndexConfig> mapIndexConfigs = mapConfig.getMapIndexConfigs();
+
+        MapIndexConfig mapIndexConfig = new MapIndexConfig();
+        mapIndexConfig.setAttribute("name");
+        mapIndexConfig.setOrdered(true);
+        mapIndexConfigs.add(mapIndexConfig);
+
+        HazelcastInstance hz1 = factory.newHazelcastInstance(config);
+        HazelcastInstance hz2 = factory.newHazelcastInstance(config);
+
+        ClientConfig clientConfig = new ClientConfig();
+        // ProxyManager#findNextAddressToSendCreateRequest uses the configured load balancer to find the next address
+        // to which proxy creation request will be sent. We want this to be member hz1.
+        clientConfig.setLoadBalancer(new StaticLB((Member) hz1.getLocalEndpoint()));
+        clientConfig.getNetworkConfig().setSmartRouting(false);
+        // the client only connects to member hz2.
+        clientConfig.getNetworkConfig().addAddress(
+                        hz2.getCluster().getLocalMember().getAddress().getHost() + ":" +
+                        hz2.getCluster().getLocalMember().getAddress().getPort());
+
+        HazelcastInstance client = factory.newHazelcastClient(clientConfig);
+
+        try {
+            IMap<String, SampleObjects.Employee> test = client.getMap("test");
+            test.put("foo", new SampleObjects.Employee(1, "name", "age", 32, true, 230));
+        } finally {
+            client.shutdown();
+            factory.shutdownAll();
```

Comment:
minor: just to make test method clean, you can shutdown factory in `@After` method. Also explicitly shutting down the client is no needed, factory handles that.